### PR TITLE
Update how 'colcon test' runs in Travis CI

### PR DESCRIPTION
**Pull request by Nickolaim**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #334 |
| Primary OS tested on | Ubuntu |
| Robotic platform tested on | N/A |

---

## Description of contribution in a few bullet points

* Before the change `colcon test` in Travis CI run in a wrong environment
* After fixing the environment, skip only packages that have test failures.  There are 11 such packages now
* Adding verbose output to `colcon test-result` so Travis CI log has errors

---

## Future work that may be required in bullet points

* I'll check issues on package by package basis and see what else I can fix.

---

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -24,10 +24,10 @@ before_install:
...
-  - docker exec nav2_bash bash -c "colcon test --packages-skip
-      dwb_controller dwb_core nav2_amcl nav2_costmap_2d nav2_dynamic_params 
-      nav2_map_server nav2_navfn_planner nav2_system_tests nav2_tasks 
-      nav2_util nav2_voxel_grid costmap_queue dwb_critics dwb_plugins 
-      nav2_bt_navigator nav2_mission_executor nav2_robot 
-      nav2_simple_navigator nav2_world_model nav_2d_utils"
-  - docker exec nav2_bash bash -c "colcon test-result" 
+  - docker exec nav2_bash bash -c '. /opt/ros/$ROS2_DISTRO/setup.bash &&
+      . /ros2_ws/ros2-linux/setup.bash &&
+      . /ros2_ws/navstack_dependencies_ws/install/setup.bash &&
```
Comment (crdelsey) [inline .travis.yml]: The last setup.bash should be the only one you need to source. They include all the setup information that was present in their dependent repos.

Comment (Nickolaim) [reply .travis.yml]: Thanks.  Addressed.

File: .travis.yml [hunk 2]
```diff
@@ -24,10 +24,8 @@ before_install:
...
 script:
   - docker run --name nav2_bash --rm -d -i -t nav2:latest bash -l
-  - docker exec nav2_bash bash -c "colcon test --packages-skip
-      dwb_controller dwb_core nav2_amcl nav2_costmap_2d nav2_dynamic_params 
-      nav2_map_server nav2_navfn_planner nav2_system_tests nav2_tasks 
-      nav2_util nav2_voxel_grid costmap_queue dwb_critics dwb_plugins 
-      nav2_bt_navigator nav2_mission_executor nav2_robot 
-      nav2_simple_navigator nav2_world_model nav_2d_utils"
-  - docker exec nav2_bash bash -c "colcon test-result" 
+  - docker exec nav2_bash bash -c '. /ros2_ws/navstack_dependencies_ws/install/setup.bash &&
```
Comment (mkhansen-intel) [inline .travis.yml]: Since this is after the navigation2_ws is built, shouldn't it be sourcing that setup.bash instead?

Comment (Nickolaim) [reply .travis.yml]: Done.

Comment (mkhansen-intel) [reply .travis.yml]: Looks good now, approved

**Pull request merged**